e2e/internal/remote/remote.go

```diff
@@ -197,3 +208,48 @@ func (c Client) Upload(localPath string, remotePath string) error {
 	log.Info("File uploaded successfully")
 	return nil
 }
+
+// Reboot reboots the remote host and waits for it to come back online, then
+// reestablishes the SSH connection.
+// It first waits for the host to go offline, then returns an error if the host
+// does not come back online within 1 minute.
+func (c *Client) Reboot() error {
+	log.Infof("Rebooting host %q", c.client.RemoteAddr().String())
+	_, _ = c.Run(context.Background(), "reboot")
+
+	waitDone := make(chan error, 1)
+	go func() {
+		waitDone <- c.client.Wait()
+	}()
+
+	// Sleep a few seconds in case SSH is still available
+	time.Sleep(10 * time.Second)
+
+	// Wait for the host to go offline
+	select {
+	case <-waitDone:
+	case <-time.After(30 * time.Second):
+		return fmt.Errorf("host did not go offline in time")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	// Wait for the host to come back online
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("host %q did not come back online after reboot", c.client.RemoteAddr().String())
+		default:
```

Comment:
You only use the context for timing out the loop here. Can't we just do ```<-time.After(time.Minute)``` instead?